public/js/pimcore/object/folder.js

```diff
@@ -214,10 +214,19 @@ pimcore.object.folder = Class.create(pimcore.object.abstract, {
             });
 
             this.toolbarButtons.remove = new Ext.Button({
-                tooltip: t('delete_folder'),
+                tooltip: t('delete'),
                 iconCls: "pimcore_material_icon_delete pimcore_material_icon",
                 scale: "medium",
-                handler: this.remove.bind(this)
+                handler: function () {
+                    var options = this.search.onRawDeleteSelectedRows();
+                    if (!options) {
+                        options = {
+                            "elementType" : "object",
+                            "id": this.id
+                        };
+                    }
```

Comment:
```suggestion
                    var options = this.search.onRawDeleteSelectedRows();
                    if (!options) {
                        options = {
                            "elementType" : "object",
                            "id": this.id
                        };
                    }
```


Also checked the case that Fash mentioned and indeed it doesn't work because the `selectionModel` is not initialized
![image](https://github.com/pimcore/admin-ui-classic-bundle/assets/6014195/904c55c0-9cb1-45c3-8bb5-95511d45d1d2)

eg. when trying to delete a folder with multiple object types and not having any object type selected, when deleting an image folder when on `preview gallery` mode (not switched to `list` view)
![image](https://github.com/pimcore/admin-ui-classic-bundle/assets/6014195/b0f64740-a0db-4486-a840-67cc4e203a39)

This code should be "tweaked", only try to delete the selected lines when found a row selection.

Reply:
Any news on this?

Reply:
Have added a check if the grid is actually loaded. If not, then the current behaviour for the button gets used (delete folder).